File: clients/php/src/NotFoundException.php
<?php

declare(strict_types=1);

namespace FileDBv2;

/**
 * Thrown when a record, key, or collection does not exist — the client mapping
 * of the gRPC NOT_FOUND status. Raised e.g. by findByKey(), updateByKey() and
 * deleteByKey() when no live record carries the requested key.
 */
class NotFoundException extends FileDBException
{
}

File: clients/java/src/main/java/com/srjn45/filedbv2/NotFoundException.java
package com.srjn45.filedbv2;

/**
 * A keyed lookup, update, or delete referenced a key that no live record holds.
 *
 * <p>Raised from {@link FileDBClient#findByKey}, {@link FileDBClient#updateByKey},
 * and {@link FileDBClient#deleteByKey} when the engine answers {@code NOT_FOUND}.
 */
public class NotFoundException extends FileDBException {

    public NotFoundException(String message, Throwable cause) {
        super(message, cause);
    }
}
